### PR TITLE
Issue #3159954 by tBKoT: Check if activity related entity exists

### PR DESCRIPTION
**Pull request by dmytriikaiun**
## Problem
Email notifications contain unconverted tokens

## Solution
Check if related entity of activity exist otherwise delete activity and exit from the queue worker process

## Issue tracker
https://getopensocial.atlassian.net/browse/YANG-3130
https://www.drupal.org/project/social/issues/3159954

### CODE REVIEW

File: modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php [hunk 1]
```diff
@@ -57,6 +67,15 @@ public static function create(ContainerInterface $container, array $configuratio
   public function processItem($data) {
     // First make sure it's an actual Activity entity.
     if (!empty($data['entity_id']) && $activity = Activity::load($data['entity_id'])) {
+      // Check if activity related entity exist.
+      if (!$activity->getRelatedEntity()) {
```
Comment (ribel) [inline modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php:71]: @dmytriikaiun There is already a function for it, so I think it is better to use it. See `\Drupal\activity_creator\ActivityNotifications::deleteNotificationsbyIds`

Comment (Kingdutch) [reply modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php:71]: Agreed with @ribel but please note there's a bug in that function that I discovered today, I have not been able to report it on Drupal.org yet.

The fix is relatively simple but also requires an update hook to delete any possibly orphaned notifications.

Comment (dmytriikaiun) [reply modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php:88]: Hi @Kingdutch 
Could you describe the bug in that function?

**Comment by ribel** [comment]
🍒 picked to 8.x and 9.x

**Pull request merged**